sycl/test-e2e/lit.cfg.py

```diff
@@ -39,12 +39,16 @@
 
 # test-mode: Set if tests should run normally or only build/run
 config.test_mode = lit_config.params.get("test-mode", "full")
+config.fallback_build_run_only = False
 if config.test_mode == "full":
     config.available_features.add("run-mode")
     config.available_features.add("build-and-run-mode")
 elif config.test_mode == "run-only":
     lit_config.note("run-only test mode enabled, only executing tests")
     config.available_features.add("run-mode")
+    if lit_config.params.get("build-instead-of-skip-run-only", False):
+        config.available_features.add("build-and-run-mode")
```

Comment:
With adding `build-and-run-mode` to `run-only` mode? 

When I was first experimenting with separating the build/run I was using features to implement the logic of the different test modes. I moved away from this because of situations like this, instead using the `config.test_mode` variable. So this shouldn't affect any of the test mode logic outside of the `ignore_line_filtering` bit added in this pr. 

AFAIK the only places the features would be used would be to evaluate UNSUPPORTED/REQUIRES/XFAIL lines. So maybe thinking about what we expect to happen in such situations? I can't think of a situation in which it would make sense to use this feature in an unsupported statement, given that `full` mode is the default testing mode. Likewise for xfail I dont think it makes sense to use the feature there either, since if a test failing on `full` mode would also have to fail in either `build-only`/`run-only` modes. That leaves the requires statements, where the addition of this I think makes sense to get around `REQUIRES: build-and-run-mode` statements, which is exactly what this pr aims to do.

Is there any other potential avenues for issues you were thinking of?